streams/early_zip_readable_streams.ts

```diff
@@ -88,21 +88,20 @@ export function earlyZipReadableStreams<T>(
   return new ReadableStream<T>({
     async start(controller) {
       try {
-        loop:
         while (true) {
           for (const reader of readers) {
             const { value, done } = await reader.read();
             if (!done) {
               controller.enqueue(value!);
             } else {
               await Promise.all(readers.map((reader) => reader.cancel()));
-              break loop;
+              controller.close();
+              break;
             }
           }
         }
-        controller.close();
-      } catch (e) {
-        controller.error(e);
+      } catch (error) {
+        controller.error(error);
       }
     },
```

Comment:
How is this not an infinite loop? The break keyword is breaking out of the for loop only to go around again in while loop, to maybe throw an error.

Could this readable stream not be better done with the pull method? That way it's not continually filling up the queue, in the stance the queue isn't being emptied.
```ts
async pull(controller) {
  for (const reader of readers) {
    const { done, value } = await reader.read()
    if (done) {
      await Promise.all(readers.map(reader => reader.cancel()))
      return controller.close()
    }
    controller.enqueue(value)
  }
}
```

Reply:
I made a mistake - reverted now. Would you like to do this in a separate performance improvement PR?